packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -135,6 +135,94 @@ class MethodChannel {
   /// * a [MissingPluginException], if the method has not been implemented by a
   ///   platform plugin.
   ///
+  /// ## Sample code
+  ///
+  /// The following code snippets demonstrate how to invoke platform methods
+  /// in Dart using a MethodChannel and how to implement those methods in Java
+  /// (for Android) and Objective-C (for iOS). The code might be packaged up as
+  /// a musical plugin, see <https://flutter.io/developing-packages/>:
+  ///
+  /// ```dart
+  /// class Music {
+  ///   static const MethodChannel _channel = const MethodChannel('music');
+  ///
+  ///   static Future<bool> isLicensed() async {
+  ///     // invokeMethod returns a Future<dynamic>, and we cannot pass that for
+  ///     // a Future<bool>, hence the indirection.
+  ///     final bool result = await _channel.invokeMethod('isLicensed');
+  ///     return result;
+  ///   }
+  ///
+  ///   static Future<List<Song>> songs() async {
+  ///     // invokeMethod deserializes compound invocation results using
+  ///     // List<dynamic> and Map<dynamic, dynamic>. Post-processing code thus
+  ///     // cannot assume e.g. List<Map<String, dynamic>> even though the
+  ///     // simple values involved would support such a typed container.
+  ///     final List<dynamic> songs = await _channel.invokeMethod('getSongs');
+  ///     return songs.map(Song.fromJson).toList();
+  ///   }
+  ///
+  ///   static Future<void> play(Song song, double volume) async {
+  ///     // Errors occurring on the platform side may be relayed to Dart
```

Comment:
I would not expose implementation details ("specially marked results") in the comment here. Can we just say: If x happens on the platform side, invokeMethod throws a PlatformException? I think that's easier to understand.